src/utils/rolePlatformHelpers.ts

```diff
@@ -0,0 +1,41 @@
+import { getTimeDiff } from "components/[guild]/RolePlatforms/components/PlatformCard/components/AvailabilityTags"
+import { RolePlatform, RolePlatformStatus } from "types"
+
+export const getRolePlatformStatus = (
+  rolePlatform: RolePlatform
+): RolePlatformStatus => {
+  const startTimeDiff = getTimeDiff(rolePlatform?.startTime)
+  const endTimeDiff = getTimeDiff(rolePlatform?.endTime)
+
+  if (
+    typeof rolePlatform?.capacity === "number" &&
+    rolePlatform?.capacity === rolePlatform?.claimedCount
+  ) {
+    return "ALL_CLAIMED"
+  } else if (startTimeDiff > 0) {
+    return "NOT_STARTED"
+  } else if (endTimeDiff < 0) {
+    return "ENDED"
+  } else {
+    return "ACTIVE"
+  }
+}
+
+export const getRolePlatformTimeframeInfo = (
+  rolePlatform: RolePlatform,
+  additionalCondition = true
+) => {
+  const startTimeDiff = getTimeDiff(rolePlatform?.startTime)
+  const endTimeDiff = getTimeDiff(rolePlatform?.endTime)
+
+  const inActiveTimeframe =
+    (startTimeDiff > 0 ||
+      endTimeDiff < 0 ||
+      (typeof rolePlatform?.capacity === "number" &&
+        rolePlatform?.capacity === rolePlatform?.claimedCount)) &&
+    additionalCondition
+
+  return { inActiveTimeframe, startTimeDiff, endTimeDiff }
```

Comment:
`inActiveTimeframe` is a bit confusing, since this variable is `true` if the reward is outside the defined timeframe, let's try another name here ;)

Reply:
You are absolutely right, thanks!
I inspected the logic again, and inverted it, so now it returns an `isAvailable` boolean instead. I also decided to remove the `additionalCondition` prop from the hook, as it takes effort to understand how it affects the return values.